src/trees/prufer.jl

```diff
@@ -0,0 +1,91 @@
+"""
+    is_tree(g)
+
+Returns true if g is a tree: that is, a simple, connected undirected graph, with nv-1 edges (nv = number of vertices). Trees are the minimal connected graphs; equivalently they have no cycles.  
+
+This function does not apply to directed graphs. Directed trees are sometimes called [polytrees](https://en.wikipedia.org/wiki/Polytree)). 
+
+"""
+
+function is_tree(g::AbstractGraph)
+    is_directed(g) && throw(ArgumentError("g must be undirected. "))
+    return is_connected(g) && ne(g) == nv(g) - 1
```

Comment:
```suggestion
    return nv(g) + 1 == ne(g)  && is_connected(g)
```
`is_connected` is much more expensive than calculating `nv(g) + 1 == ne(g)`  so I would do the second calculation first, if that one is false, then we don't need to check for connectivity.